docs/tutorials/Remark.adoc

```diff
@@ -0,0 +1,363 @@
+            + PREFIX_REMARK + "Likes to swim.";
+
+    public static final String MESSAGE_NOT_IMPLEMENTED_YET = "Remark command not implemented yet";
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        throw new CommandException(MESSAGE_NOT_IMPLEMENTED_YET);
+    }
+}
+----
+
+NOTE: Remember to add `PREFIX_REMARK` to the `CliSyntax` class!
+
+Check out the code for the `RemarkCommand`
+link:++https://github.com/j-lum/addressbook-level35/commit/72256eacea79dfcf077f5c7cb6f89819c02f090e#diff-34ace715a8a8d2e5a66e71289f017b47++[here].
+Feel free to copy and paste this.
+
+=== Hook `RemarkCommand` into the application
+
+Now that we have our `RemarkCommand` ready to be executed, we need to update `AddressBookParser#parseCommand()` to
+recognize the `remark` keyword. Add the new command to the switch-case and return a new instance of `RemarkCommand`.
+
+You can refer to the changes in this
+link:++https://github.com/j-lum/addressbook-level35/commit/72256eacea79dfcf077f5c7cb6f89819c02f090e#diff-5338391f3f6fbb4022c44add6590b74f++[diff].
+
+=== Run the application
+
+Run `Main#main` and try out your new `RemarkCommand`. If everything went well, you should see something like this:
+
+.Successful failure
+image::RemarkNotImplemented.png[]
+
+== Enhancing `RemarkCommand`
+
+While we have gotten AddressBook to recognize our new command, it does not parse input any input from the user. Let's
+ change that.
+
+=== Make the command accept parameters
+
+We start by modifying the constructor of `RemarkCommand` to accept an `Index` and a `String`. While we are at it,
+let's change the error message to echo the values. While this is not a replacement for tests, it is an obvious way to
+ tell if our code is functioning as intended.
+
+[source, java]
+----
+private final Index index;
+private final String remark;
+
+/**
+ * @param index of the person in the filtered person list to edit the remark
+ * @param remark of the person to be updated to
+ */
+public RemarkCommand(Index index, String remark) {
+    requireAllNonNull(index, remark);
+
+    this.index = index;
+    this.remark = remark;
+}
+@Override
+public CommandResult execute(Model model) throws CommandException {
+    throw new CommandException(String.format(MESSAGE_ARGUMENTS, index.getOneBased(), remark));
+}
+----
+
+Your code should look something like
+link:++https://github.com/j-lum/addressbook-level35/commit/c0f8419be4fb59bffafcd73026769459e5cf5c5e#diff-34ace715a8a8d2e5a66e71289f017b47++[this] after you are done.
+
+=== Parse user input
+
+Now let's move on to writing a parser that will extract the index and remark from the input provided by the user.
+
+Create a `RemarkCommandParser` class in the `seedu.address.logic.parser` package. The class must extend the `Parser`
+interface.
+
+.UML diagram describing the relationship between Parser and RemarkCommandParser
+image::ParserInterface.png[width=386px, height=350px]
```

Comment:
![image](https://user-images.githubusercontent.com/1673303/63442395-9e6a0500-c465-11e9-8965-dd01527b430a.png)

Notation issues

Reply:
Do we have a recommended notation for exceptions and generics?

Reply:
We don't cover either. Perhaps omit them from the diagram as well?